spec/controllers/users/devices_controller_spec.rb

```diff
@@ -147,6 +147,22 @@
       expect(device.privilege_for(developer)).to_not be priv
     end
 
+    it 'should switch privilege for a developer on all devices' do
+      developer = FactoryGirl::create(:developer)
+      device.developers << developer
+      device.user = user
+      device.save
+      priv = device.privilege_for(developer)
+
+      request.accept = 'text/javascript'
+      post :switch_all_privileges_for_developer, {
+        user_id: user.username,
+        developer: developer.id
+      }
+
+      expect(device.privilege_for(developer)).to_not be priv
```

Comment:
This test probably ought to test that it isn't just happening to one device, but all of them :)